### ISSUE TITLE
Ease database diagnosis via local IP access configuration

### ISSUE THREAD

**Issue by phillxnet** [rockstor/rockstor-core#2730]
We currently limit our Host Based Authentication (`./conf/pg_hba.conf`) to what is required functionally and for running our tests. That is unix socket access only. It is proposed that we add the following, **but remarked out**, configurations that enable local IP host access that can be invaluable while working on database related issues and using for example a remove PyCharm setup where a IDE client exists locally on the development VM instance:

```
host   storageadmin    rocky              127.0.0.1/32            md5
host   smartdb         rocky              127.0.0.1/32            md5
host   storageadmin    rocky              ::1/128                 md5
host   smartdb         rocky              ::1/128                 md5
```

**Comment by phillxnet** [comment]
Our existing access is via local Unix socket only. And it is not proposed that we change this, only that we add the above as a way to ease folks ability to enable IP based access from localhost (IPv4 & IPv6).

**Comment by Hooverdan96** [comment]
@phillxnet, this additional "option" would then also find entry into our dev contribution docs or is that something an experienced developer would automatically hunt for?

**Comment by phillxnet** [comment]
@Hooverdan96 

> find entry into our dev contribution docs

Possibly, yes. But when trying to do this the error indicates a refusal based on HBA auth so looking at the file and finding a solution is within the remit of those editing such a file. Just wanted to make it easier once one gets to that file. We don't want to reproduce all upstream docs, and folks messing with hba is fairly deep. But yes over-all - i.e. I'd like a PyCharm pro setup HowTo  in the end, and these entries already in-config (but remarked out) simplifies the on-boarding bit. So we could have a quick reference within this proposed PyCharm remote setup howto was my thinking. We don't otherwise have docs relating to this level of development as it goes. And I'm not too keen on establishing a doc section with cli approaches to doing all this db stuff actually as it would be another book. Where-as point and click to see this or the other table-db is super handy within the IDE setting, and accessible from the get-go as it were. But only in Pro PyCharm unfortunately.

**Comment by Hooverdan96** [comment]
the PyCharm pro setup HowTo would be a great idea I think!

**Comment by phillxnet** [comment]
Closing as:
Fixed by #2736
